js/ui/windowManager.js

```diff
@@ -189,6 +206,20 @@ var EOSShellWobbly = new Lang.Class({
 
         actor.disconnect(this._positionChangedId);
         this._positionChangedId = null;
+
+        // Remove the effect from the actor if necessary
+        if (!this.enabled) {
+            this._removeFromActorOnDisabled(actor);
+        }
+
+        // Watch for the effect to be disabled and remove if it
+        // it does get disabled - saves some CPU cycles computing
+        // things like boundings boxes
+        if (this._wobblyEffectDisabledId === -1) {
+            this._wobblyEffectDisabledId = this.connect('notify::enabled', Lang.bind(this, function() {
```

Comment:
You could also do this with a vfunc, which would save you the whole signal connect/disconnect boilerplate.

Reply:
Which vfunc would this be?

Reply:
`vfunc_notify` I believe